lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -336,18 +341,19 @@ def revert_json_html_encoding!(string)
       def api_request(method, url, rel_path, params)
         params == {} ? body = '' : body = params.to_json
         revert_json_html_encoding!(body) if defined?(ActiveSupport::JSON::Encoding) && ActiveSupport::JSON::Encoding.escape_html_entities_in_json
-        parse(ssl_request(method, url, body, headers(rel_path, body)))
+        parse(ssl_request(method, url, body, headers(rel_path, body, params)))
       end
 
-      def headers(rel_path, payload)
+      def headers(rel_path, payload, params)
         salt = SecureRandom.base64(12)
         timestamp = Time.new.to_i.to_s
         {
           'Content-Type' => 'application/json',
           'access_key' => @options[:access_key],
           'salt' => salt,
           'timestamp' => timestamp,
-          'signature' => generate_hmac(rel_path, salt, timestamp, payload)
+          'signature' => generate_hmac(rel_path, salt, timestamp, payload),
+          'idempotency' => @options[:idempotency]
```

Comment:
Is this value allowed to be nil?

Reply:
good point, if we don't set a value for idempotency we should not send it, tahnks, I'll update it